grype/pkg/package.go

```diff
@@ -101,21 +103,22 @@ func (p Package) String() string {
 	return fmt.Sprintf("Pkg(type=%s, name=%s, version=%s, upstreams=%d)", p.Type, p.Name, p.Version, len(p.Upstreams))
 }
 
-func removePackagesByOverlap(catalog *pkg.Collection, relationships []artifact.Relationship) *pkg.Collection {
+func removePackagesByOverlap(sbm *sbom.SBOM) *pkg.Collection {
+	catalog := sbm.Artifacts.Packages
 	byOverlap := map[artifact.ID]artifact.Relationship{}
-	for _, r := range relationships {
+	for _, r := range sbm.Relationships {
 		if r.Type == artifact.OwnershipByFileOverlapRelationship {
 			byOverlap[r.To.ID()] = r
 		}
 	}
 
 	out := pkg.NewCollection()
-
+	comprehensiveDistroFeed := distroFeedIsComprehensive(sbm.Artifacts.LinuxDistribution)
```

Comment:
It seems a little odd to add this call here rather than, say, passing the sbom to `excludePackages` on line 121

Reply:
the nice thing about calling it here though is that `excludePackage` is minimizing what it needs as inputs.